### ISSUE TITLE
[BUG] Inconsistent firing of melee/ranged swings on bosses with large hitboxes

### ISSUE THREAD

**Issue by MannySJH** [ClassicWoWCommunity/cata-classic-bugs#1381]
### 1. Current Behavior

1.1. Bosses with large hitboxes are causing hunters in close proximity to begin autoattacking with their melee weapons instead of firing autoshots. During this you can still use all ranged spells (5 to X yards) whilst simultaneously being able to use raptor strike (Melee range)

  #### 1.3  Source Material
As you can see in [this](https://streamable.com/jofjja) clip, I'm able to use my ranged attacks while in "melee" range but my autoshot does not fire. I move back a few yards and normal damage continues. Notice the raptor strike becoming available and unavailable on the action bars to indicate this range. 

### 2. Expected Behavior
In [this](https://youtu.be/nCtVRsiX_mo?si=1aioG9RYHTU-Iaxy&t=23) video from 2011 on the same boss you can see the raptor strike is not available to use on the hunters bars, the boss is then moved closer to the hunter and raptor becomes available indicating "melee" range. However the hunters autoshots continue to fire contradictory to the clip I recorded on current live servers. 
  
During the air phase of Al'Akir it's almost impossible to escape this huge hybrid zone resulting in a massive dps loss for hunters, note as well that melee attacks do not fire during casting of cobra/steady shots so the autoshot damage isn't even substituted for melee damage. I have no footage of this on live but in [this](https://youtu.be/UDL32kBPigs?si=OrMZ1UkoNxsDjeoW&t=294) video you can see the hunter firing their autoshots in this phase where on live servers currently this isn't possible. 

I also did manage to [find](https://youtu.be/3MQyH78aqT4?si=YQjrtmdaO1zS12E7&t=660) footage of the melee swing firing instead of the ranged swing however the range requirements for this seem to be more exaggerated right now compared to back then, regardless this is extremely frustrating for hunters to deal with in raids.

**Comment by MannySJH** [comment]
I think I isolated the problem to this setting 
![Capture](https://github.com/JamminL/cata-classic-bugs/assets/168474168/b782e3cc-40d6-4e5d-a90c-dbb02c462d10)

All testing prior to this addendum was done with this setting turned on since it's been the norm for hunters since wotlk. It makes sure autos fly without being toggled off, melee swings starts when someone enters your melee zone and allows you to right click to fire autos in a pinch, like on totems for example. It also circumvents a few other issues.

Having this setting off brings up a multitude of problems such as delayed auto swing timers, auto shot being toggled off despite the exclamation mark before /cast auto shot, cancelling of steady/cobra casts which completely negates any spell queuing quality of life and others I'm probably not even aware of. 

For the purposes of testing I did toggle this setting off and [found](https://streamable.com/l023yv) that the problem above was completely gone although some of the issues above came to the forefront also. I believe the client is prioritizing melee swings instead of ranged swings in this hybrid zone, hopefully this can be changed/fixed without impacting how the setting works outside of certain cataclysm bosses. Note that this hybrid melee/ranged zone doesn't exist on any bosses prior to cataclysm which leads me to believe it was added as a quality of life addition specifically for hunters for bosses with larger hitboxes.

**Comment by MannySJH** [comment]
Made another discovery today, setting Auto Attack/Auto Shot on with a combination of /cast !auto shot in macros instead of /startattack also fixes the problem. I was able to fire my autos in this hybrid zone, however repeated spam of said macro will cancel any casts such as cobra or steady so the solution is still not ideal.

**Comment by MannySJH** [comment]
We've [found](https://discord.com/channels/253213664708919296/1171712100826087454/1249532954795970631) a way around this in the hunter discord but it's far from ideal and still extremely janky. Hopefully this helps in fixing the problem properly.

**Comment by MannySJH** [comment]
Since the last time I updated this thread we've made a lot of headway as to why and how the issues presented occur. We also figured out that even with our unorthodox solution a player entering the problem zone can inadvertently stop our "fix" from having  any effect based on which swing was first cast on the boss in said zone. Can read more [here](https://discord.com/channels/253213664708919296/1249128902413254758). This is extremely frustrating for people who might not fully understand why this happens and probably even more so if the player in question isn't part of the hunter discord. The simplest solution from what I can gather is having an option to completely suppress any kind of automatic melee swing whilst simultaneously having autorangedcombat set to 1. Something like an additional option box that appears when you tick the original and maybe defaulted to on. With weakauras we can also automate a cvar toggle on problem bosses which shouldn't effect any other gameplay meaning the regular interaction should remain completely unchanged.
![Untitled](https://github.com/ClassicWoWCommunity/cata-classic-bugs/assets/168474168/a83edbda-86c7-4880-9b49-5347864932cf)
